irobot_benchmark/CMakeLists.txt

```diff
@@ -24,20 +24,90 @@ install(TARGETS
   DESTINATION lib/${PROJECT_NAME}
 )
 
-set(TOPOLOGY_FILES
-  topology/sierra_nevada.json
-  topology/cedar.json
-  topology/mont_blanc.json
-  topology/white_mountain.json
-  topology/debug_sierra_nevada_reliable.json
-  topology/debug_sierra_nevada_best_effort.json
-  topology/debug_mont_blanc_reliable.json
-  topology/debug_mont_blanc_best_effort.json
+set(FT topology/full_topologies)
+
+set(TEST_TOPOLOGY_FILES
+  ${FT}/cedar.json
+  ${FT}/debug_mont_blanc_best_effort.json
+  ${FT}/debug_mont_blanc_reliable.json
+  ${FT}/debug_sierra_nevada_best_effort.json
+  ${FT}/debug_sierra_nevada_reliable.json
+  ${FT}/debug_white_mountain_best_effort.json
+  ${FT}/mont_blanc.json
+  ${FT}/sierra_nevada.json
+  ${FT}/sierra_nevada_fixed_size.json
+  ${FT}/debug_sierra_nevada_fixed_size_best_effort.json
+  ${FT}/sierra_nevada_fixed_size_loaned.json
+  ${FT}/sierra_nevada_loaned.json
+  ${FT}/white_mountain.json
+  ${FT}/white_mountain_fixed_size.json
+  ${FT}/debug_white_mountain_fixed_size_best_effort.json
+  ${FT}/white_mountain_fixed_size_loaned.json
+  ${FT}/white_mountain_loaned.json
+)
+
+install(FILES
+  ${TEST_TOPOLOGY_FILES}
+  DESTINATION lib/${PROJECT_NAME}/${FT})
+
+set(SP topology/single_process)
+
+set(TOPOLOGY_FILES_SINGLE_PROCESS
+  ${SP}/pub_sub_loaned_10b.json
+  ${SP}/pub_sub_loaned_100kb.json
+  ${SP}/pub_sub_loaned_1mb.json
+  ${SP}/pub_sub_loaned_4mb.json
+
+  ${SP}/pub_sub_10b_ipc_on.json
+  ${SP}/pub_sub_100kb_ipc_on.json
+  ${SP}/pub_sub_1mb_ipc_on.json
+  ${SP}/pub_sub_4mb_ipc_on.json
+
+  ${SP}/pub_sub_10b_ipc_off.json
+  ${SP}/pub_sub_100kb_ipc_off.json
+  ${SP}/pub_sub_1mb_ipc_off.json
+  ${SP}/pub_sub_4mb_ipc_off.json
 )
 
 install(FILES
-  ${TOPOLOGY_FILES}
-  DESTINATION lib/${PROJECT_NAME}/topology)
+  ${TOPOLOGY_FILES_SINGLE_PROCESS}
+  DESTINATION lib/${PROJECT_NAME}/${SP})
+
+set(MP topology/multi_process)
+
+set(TOPOLOGY_FILES_MULTI_PROCESS
+  ${MP}/pub_ipc_off_10b.json
+  ${MP}/pub_ipc_off_100kb.json
+  ${MP}/pub_ipc_off_1mb.json
+  ${MP}/pub_ipc_off_4mb.json
+
+  ${MP}/sub_ipc_off_10b.json
+  ${MP}/sub_ipc_off_100kb.json
+  ${MP}/sub_ipc_off_1mb.json
+  ${MP}/sub_ipc_off_4mb.json
+
+  ${MP}/pub_loaned_10b.json
+  ${MP}/pub_loaned_100kb.json
+  ${MP}/pub_loaned_1mb.json
+  ${MP}/pub_loaned_4mb.json
+
+  ${MP}/sub_loaned_10b.json
+  ${MP}/sub_loaned_100kb.json
+  ${MP}/sub_loaned_1mb.json
+  ${MP}/sub_loaned_4mb.json
+)
+
+install(FILES
```

Comment:
can't we install directories rather than files?
this would allow to not have to list them all here

Reply:
It makes sense to me, I agree these kind of resources usually live under the `share` directory structure.

@mauropasse FYI

Reply:
@alsora , by doing as suggested, these resources get installed, respectively, under:
```
share/profiles
share/single_process
share/full_topologies
share/multi_process
```

OTOH, if installed with something like:
```
install(
  DIRECTORY
    profiles
    topology
  DESTINATION
    share/${PROJECT_NAME}
)
```
The installation paths become:
```
share/irobot_benchmark/profiles
share/irobot_benchmark/topologies/single_process
share/irobot_benchmark/topologies/full_topologies
share/irobot_benchmark/topologies/multi_process

```

which to my eyes look more organized. Please let me know your thoughts on this.

Reply:
I'm not sure if we need `share/${PROJECT_NAME}` as destination.

The `PROJECT_NAME` directory may already be included depending on how you `colcon-build` the project (i.e. with or without `--merge-install` flag)